Castle/Classes/Client/CASEvent.m

```diff
@@ -84,13 +84,21 @@ - (id)JSONPayload
 {
     NSString *timestamp = [[CASModel timestampDateFormatter] stringFromDate:self.timestamp];
     NSDictionary *context = @{ @"device": [[CASDevice sharedDevice] JSONPayload] };
-
-    NSMutableDictionary *payload = @{ @"type": self.type,
-                                      @"event": self.name,
-                                      @"properties": self.properties,
-                                      @"timestamp": timestamp,
-                                      @"context": context }.mutableCopy;
-
+    NSMutableDictionary *payload = nil;
+
+    if ([self.type isEqualToString:@"screen"]) {
```

Comment:
@aalekz we probably should have a separate method on the Screen event instead of doing this ugly check here. That refactoring was beyond my knowledge.